org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/prefs/ICustomClientPreferenceDo.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2010-2022 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.shared.prefs;
+
+import org.eclipse.scout.rt.dataobject.IDoEntity;
+import org.eclipse.scout.rt.dataobject.TypeVersionRequired;
+
+/**
+ * A marker interface for custom client preferences.
+ */
+@TypeVersionRequired
+public interface ICustomClientPreferenceDo extends IDoEntity {
```

Comment:
What are the rules regarding the correct module? Do implementations of ICustomClientPreferenceDo always have to be in "shared" modules? Users might be confused because the ClientUIPreferences API itself is located in the "client" module, and it has the word "client" in its name. Maybe this can be documented here.